client/src/unifycr-sysio.c

```diff
@@ -1400,6 +1408,21 @@ int unifycr_fd_logreadlist(read_req_t *read_req, int count)
     int *ptr_size = NULL;
     int *ptr_num = NULL;
 
+    /*
+     * Adjust length for fitting the EOF.
+     */
+    for (i = 0; i < count; i++) {
```

Comment:
haven't parsed this code to fully understand it, but can this be done without a for loop?

Reply:
Nevermind, I see now that we're dealing with an array of requests.